### ISSUE TITLE
mutationThreshold as 100 is not guaranty that all classes is covered by 100%

### ISSUE THREAD

**Issue by romani** [hcoles/pitest#375]
steps to reproduce
```
$ git clone git@github.com:checkstyle/checkstyle.git
$ git checkout 3df996acc3bfc8072bcb5ce50009add08c3104dc
$ mvn -Ppitest-checks-annotation,no-validations clean verify org.pitest:pitest-maven:mutationCoverage -Dverbose=true
```

![image](https://user-images.githubusercontent.com/812984/28281515-3cd06fda-6adc-11e7-8ad2-e85e43435e7c.png)

Unfortunately it could lead to leek of incomplete test even on CI level is demanded to be 100% for Mutation Coverage `<mutationThreshold>100</mutationThreshold>` .

Expected: when some class is less then 100%, summary mutation coverage should be less then 100% and to have build failed if `<mutationThreshold>100</mutationThreshold>` present in config.

**Comment by romani** [comment]
the same problem for `coverageThreshold`, 100 in config does not trigger build failure when come class is covered a bit lower than 100.
Post processing of report by "grep/...." is required to enforce 100 always, and have no leak.

**Comment by nicolasSoyeur** [comment]
@romani I have the same issue sometimes. My team is alway putting the threshold to 100. I found a round in the source code of the plugin, which put the percentage to 100% even if you have mutant still alive.
e.g : you have 500 mutant generated, but 499 killed, the pluging compute a Math.round((100 / 500) * 499) = Math.round(99.8) = 100

 I don't know for the coverageeThreshold. I will check later

**Comment by romani** [comment]
Is it same as https://github.com/hcoles/pitest/pull/638 ?

**Comment by nicolasSoyeur** [comment]
Ohh sounds like there are conflict about rounding down and up :(

**Comment by romani** [comment]
@Kevin222004, please send PR to update pitest to latest version

**Comment by Vampire** [comment]
Ah, no, now I can say my PR was not related. :-D
But probably the same logic should be used:
![image](https://github.com/hcoles/pitest/assets/325196/160061c7-9fcf-4243-966a-99a13f9d611a)

**Comment by Vampire** [comment]
#1221 should finally fix it hopefully though

**Comment by Vampire** [comment]
It was merged, so next release after 1.14.1 should fix it.
